File: C/identity_matrix_check.c
#include <stdio.h>

int isIdentity(int mat[100][100], int n) {
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            if ((i == j && mat[i][j] != 1) || (i != j && mat[i][j] != 0))
                return 0;
    return 1;
}

int main() {
    int n, mat[100][100];
    printf("Enter size of square matrix: ");
    scanf("%d", &n);

    printf("Enter matrix:\n");
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            scanf("%d", &mat[i][j]);

    if (isIdentity(mat, n))
        printf("Identity Matrix\n");
    else
        printf("Not an Identity Matrix\n");
    return 0;
}

File: Python/identity_matrix_check.py
def is_identity(matrix):
    size = len(matrix)
    for i in range(size):
        for j in range(size):
            if (i == j and matrix[i][j] != 1) or (i != j and matrix[i][j] != 0):
                return False
    return True

# Example
identity = [
    [1, 0, 0],
    [0, 1, 0],
    [0, 0, 1]
]

print("Is identity matrix:", is_identity(identity))  # Output: True
